// type-checks/dist/es/index.js
export const isUndef = (val) => typeof val === "undefined";
export const isNull = (val) => val === null;
export const isNullOrUndef = (val) => isNull(val) || isUndef(val);
export const isString = (val) => typeof val === "string" || val instanceof String;
export const isNumber = (val) => typeof val === "number" && !isNaN(val);
export const isNumberOrString = (val) => isNumber(val) || isString(val);
export const isBool = (val) => typeof val === "boolean";
export const isFunction = (val) => typeof val === "function";
export const isDate = (val) => Object.prototype.toString.call(val) === "[object Date]";
export const isFormatDate = (val) => /^\d{4}-\d{2}-\d{2}$/g.test(val);
export const isFormatDateTime = (val) => /^\d{4}-\d{2}-\d{2}\s\d{2}:\d{2}:\d{2}$/g.test(val);
export const isFormatDateUTC = (val) => /^\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}Z$/g.test(val);
export const isObject = (value) => value !== null && typeof value === "object";
export const isEmpty = (val) => {
    if (isNullOrUndef(val)) {
        return true;
    }
    return (Object.keys(val).length === 0 && val.constructor === Object)
        || (Array.isArray(val) && val.length === 0)
        || (isString(val) && val.length === 0);
};
