// type-checks/dist/index.js
"use strict";
exports.isUndef = (val) => typeof val === "undefined";
exports.isNull = (val) => val === null;
exports.isNullOrUndef = (val) => exports.isNull(val) || exports.isUndef(val);
exports.isString = (val) => typeof val === "string" || val instanceof String;
exports.isNumber = (val) => typeof val === "number" && !isNaN(val);
exports.isNumberOrString = (val) => exports.isNumber(val) || exports.isString(val);
exports.isBool = (val) => typeof val === "boolean";
exports.isFunction = (val) => typeof val === "function";
exports.isDate = (val) => Object.prototype.toString.call(val) === "[object Date]";
exports.isFormatDate = (val) => /^\d{4}-\d{2}-\d{2}$/g.test(val);
exports.isFormatDateTime = (val) => /^\d{4}-\d{2}-\d{2}\s\d{2}:\d{2}:\d{2}$/g.test(val);
exports.isFormatDateUTC = (val) => /^\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}Z$/g.test(val);
exports.isObject = (value) => value !== null && typeof value === "object";
exports.isEmpty = (val) => {
    if (exports.isNullOrUndef(val)) {
        return true;
    }
    return (Object.keys(val).length === 0 && val.constructor === Object)
        || (Array.isArray(val) && val.length === 0)
        || (exports.isString(val) && val.length === 0);
};
